easybuild/easyblocks/s/superlu.py

```diff
@@ -0,0 +1,148 @@
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing the SuperLU library, implemented as an easyblock
+
+@author: Xavier Besseron (University of Luxembourg)
+"""
+
+import os
+from distutils.version import LooseVersion
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+from easybuild.tools.modules import get_software_root,get_software_version
+
+
+class EB_SuperLU(CMakeMake):
+    """
+    Support for building the SuperLU library
+    """
+
+    @staticmethod
+    def extra_options():
+        """
+        Define custom easyconfig parameters for SuperLU.
+        """
+        extra_vars = {
+            'build_shared_libs': [False, "Build shared library (instead of static library)", CUSTOM],
+        }
+        return CMakeMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """
+        Set the CMake options for SuperLU
+        """
+        self.cfg['separate_build_dir'] = True
+
+        if self.cfg['build_shared_libs']:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=ON')
+        else:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=OFF')
+
+        # Add -fPIC flag if necessary
+        if self.toolchain.options['pic']:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=ON')
+        else:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=OFF')
+
+        # Make sure not to build the slow BLAS library included in the package
+        self.cfg.update('configopts', '-Denable_blaslib=OFF')
+
+        # Set the BLAS library to use
+        # For this, use the BLA_VENDOR option from the FindBLAS module of CMake
+        # Check for all possible values at https://cmake.org/cmake/help/latest/module/FindBLAS.html
+        toolchain_blas = self.toolchain.definition().get('BLAS', ['NO_BLAS_IN_TOOLCHAIN'])[0]
```

Comment:
my mistake, let's not use a random string like `'NO_BLAS_IN_TOOLCHAIN'`, use `None` instead (also below)